### DISCUSSION TITLE
Max purchase quantity + permanent items

### DISCUSSION THREAD

**Discussion by jmmullins** [jeremyckahn/farmhand#437] [Ideas]
I'm writing this fully aware that we probably don't have custom purchase-quantity logic for ANY items in the shop, but I think the game could benefit from some additional parameters on certain items...

1. Sprinkler - I know the quantity for 1/X (where X is max purchasable quantity) seems to be currently based on a combo of available inventory space + money available, but considering that sprinklers never degrade/get broken/are lost and the MAX field size only requires 24 sprinklers at the smallest sprinkler coverage range.. could we add an additional argument for it to do the calculation and if X purchase quantity would be over 24 it just caps at 24? If less than, obviously show the actual number, but it doesn’t quite make sense to see **"1/890"** when the field only requires 24 (or less as sprinklers get stronger). Even if i was a bit of a psychopath and wanted sprinklers on EVERY plot that’s still maxed at 216. Also as sprinkler range increases, the quantity required goes down significantly, but I'd still rather see "1/24" on a field that only needs 8 sprinklers than "1/890" or whatever. 
2. Hug machines - similar to sprinklers, you can only ever need 30 max, or less based on pen size.

Adding those arguments so that max count doesn't exceed what you can use would be fantastic, but going one step further and having it be smart enough to know how many I already have (in use AND in inventory) in order to do dynamically calculate to show MAX minus OWNED so that the max purchasable quantity is still never over what I'll be able to use in a lifetime. 
It's conceivable that a player may not purchase 30 hug machines to start with when they only have a few cows, but if we only make the initially suggested change that player may end up still seeing "1/30" in the shop even though they already have 5 let's say instead of "1/25".

The thought behind this request is to avoid accidentally over-buying items I know I can never use that many of (and then having to sell them at half the price I just paid). I recognize this not really a problem in the current game, but it would be very nice for it to just "know" and not me having to think about it.

Seeds and other expendable items make sense to leave as-is / allow you to purchase more than you could use at one time, this is primarily focusing on the "forever" items (unsure if there are others later in the game that would have similar inherent maximums). 

Food for thought. This topic got a bit more ramble-y, let me know if anything needs clarification

**Comment by jeremyckahn** [comment]
I think this idea makes a lot of sense for Hugging Machines (you'll never need more than 30), but it doesn't feel quite right to limit Sprinkler purchasing. They're a bit more flexible in nature (their range expands as you level up) and a limit assumes that the player is going to place them efficiently. IMO it would be safer to set the limit to the maximum amount of plots (216), but that also seems a little off to me as the limit may come across as arbitrary to players.

For Sprinklers, I think it would be more interesting to have them degrade and disappear as you mentioned in #436. It would eliminate the need for a limit and also make the game more challenging. I'd like to explore that path instead, so I've filed #441 to get it done. 🙂
